src/nvim/ex_getln.c

```diff
@@ -201,7 +201,7 @@ static Array cmdline_block = ARRAY_DICT_INIT;
 /*
  * Type used by call_user_expand_func
  */
-typedef void *(*user_expand_func_T)(char_u *, int, char_u **, int);
+typedef void *(*user_expand_func_T)(const char_u *, const int, const char_u * const * const, const _Bool);
```

Comment:
Only `const` behind a pointer is meaningful in a signature, so: `(const char_u *, int, const char_u *const *, bool)`